Lib/urllib/parse.py

```diff
@@ -998,27 +1061,59 @@ def splitnport(host, defport=-1):
         return host, nport
     return host, defport
 
-def splitquery(url):
+
+def splitquery(url=''):
+    warnings.warn("urllib.parse.splitquery() is deprecated as of 3.7, "
+                  "use urllib.parse.urlparse() instead",
+                  DeprecationWarning, stacklevel=2)
+    return _splitquery(url)
+
+
+def _splitquery(url):
     """splitquery('/path?query') --> '/path', 'query'."""
     path, delim, query = url.rpartition('?')
     if delim:
         return path, query
     return url, None
 
-def splittag(url):
+
+def splittag(url=''):
+    warnings.warn("urllib.parse.splittag() is deprecated as of 3.7, "
+                  "use urllib.parse.urlparse() instead",
+                  DeprecationWarning, stacklevel=2)
+    return _splittag(url)
+
+
+def _splittag(url):
     """splittag('/path#tag') --> '/path', 'tag'."""
     path, delim, tag = url.rpartition('#')
     if delim:
         return path, tag
     return url, None
 
-def splitattr(url):
+
+def splitattr(url=''):
+    warnings.warn("urllib.parse.splitattr() is deprecated as of 3.7, "
+                  "use urllib.parse.urlparse() instead",
+                  DeprecationWarning, stacklevel=2)
+    return _splitattr(url)
+
+
+def _splitattr(url):
     """splitattr('/path;attr1=value1;attr2=value2;...') ->
         '/path', ['attr1=value1', 'attr2=value2', ...]."""
     words = url.split(';')
     return words[0], words[1:]
 
-def splitvalue(attr):
+
+def splitvalue(url=''):
+    warnings.warn("urllib.parse.splitvalue() is deprecated as of 3.7, "
+                  "use urllib.parse.urlparse() instead",
```

Comment:
Thank you.  You're right, urlparse wasn't a good substitute.  I was thinking that the `query` part of the result from urlparse would be similar to splitvalue, but that's closer to splitquery and still would need extra parsing.   I've changed it to `use urllib.parse.parse_qsl`.